docs/access/stratum1.md

```diff
@@ -0,0 +1,357 @@
+
+* Install the `cvmfs-server` package;
+* Add the public key of the CernVM-FS repository you want to replicate to `/etc/cvmfs/keys/`;
+* Create the repository replica;
+* Run the initial synchronisation;
+* Configure `cron` to perform periodic synchronisation;
+
+In the sections below, we will set up a Stratum 1 replica server for the [EESSI CernVM-FS repository `software.eessi.io`](../eessi/high-level-design.md#filesystem_layer).
+
+### Installing CernVM-FS server
+
+Start with installing the `cvmfs-server` package which provides the CernVM-FS client component.
+
+Although we won't actually use the functionality that requires it,
+we also need to install a package that provides the `mod_wsgi` Apache adapter module.
+
+=== "For RHEL-based Linux distros (incl. CentOS, Rocky, Fedora, ...)"
+
+    ``` { .bash .copy }
+    # install cvmfs-release package to add yum repository
+    sudo yum install -y https://ecsft.cern.ch/dist/cvmfs/cvmfs-release/cvmfs-release-latest.noarch.rpm
+
+    # install CernVM-FS server package
+    sudo yum install -y cvmfs-server
+
+    # install mod_wsgi Apache adapter module
+    # (on versions older than equivalent to RHEL8, install mod_wsgi instead)
+    sudo yum install -y python3-mod_wsgi
+    ```
+
+=== "For Debian-based Linux distros (incl. Ubuntu)"
+
+    ``` { .bash .copy }
+    # install cvmfs-release package to add apt repository
+    sudo apt install lsb-release
+    curl -OL https://ecsft.cern.ch/dist/cvmfs/cvmfs-release/cvmfs-release-latest_all.deb
+    sudo dpkg -i cvmfs-release-latest_all.deb
+    sudo apt update
+
+    # install CernVM-FS server package + the required mod-wsgi
+    sudo apt install -y cvmfs-server
+
+    # install mod_wsgi Apache adapter module
+    # (on Ubuntu versions older than 22.04, install libapache2-mod-wsgi instead)
+    sudo apt install -y libapache2-mod-wsgi-py3
+    ```
+
+### Adding repository public key
+
+Add the public key for the repositories in the `eessi.io` domain to `/etc/cvmfs/keys`:
+
+``` { .bash .copy }
+sudo mkdir -p /etc/cvmfs/keys/eessi.io/
+sudo cp eessi.io.pub /etc/cvmfs/keys/eessi.io/
+```
+
+You can get the contents for `eessi.io.pub` from the default CernVM-FS config repository on a CernVM-FS client system
+at:
+
+``` { .bash .copy }
+/cvmfs/cvmfs-config.cern.ch/etc/cvmfs/keys/eessi.io/eessi.io.pub
+```
+
+### Creating repository replica
+
+To create the repository replica, we need to use run the `cvmfs_server add-replica` command.
+
+#### Synchronisation server
+
+We will need to specify the server that should be used for synchronising
+the repository contents.
+This can either be the [Stratum 0 server](../appendix/terminology.md#stratum0), or a public Stratum 1 replica server
+that was set up to be used for repository synchronisation
+(by having a `.cvmfs_master_replica` file in the HTTP root directory).
+
+For EESSI we should use `aws-eu-west-s1-sync.eessi.science` as synchronisation server.
+
+#### Disabling the GeoAPI
```

Comment:
I'm not entirely sure how clients usually pick the initial Stratum 1 and request it to do the geographical ordering (is it random or the first one in the list that is available), and hence I'm not sure if this is a good recommendation. Maybe @jblomer could give some feedback here if this approach works. You basically want to always prefer the local Stratum one, unless it's not available, but I'm not sure if that's possible?

Reply:
Now that I think of it again, perhaps it will work in any case: if it picks the local/private Stratum 1 first, and it doesn't have GeoAPI enabled, perhaps it will just use that one anyway? And if it does go to a public Stratum 1 first, that will have GeoAPI enabled, and therefore it should redirect the client back to the local/private Stratum 1 (assuming it figures out that this is the geographically closest one)?

Reply:
Currently, the clients expect all or none of the stratum 1s to support GeoAPI. If GeoAPI is enabled in the client configuration, the client will pick a random stratum 1 and ask it to order the list of stratum 1s geographically. The client then uses the (re-)ordered list.

If the Geo-API request fails, the client will go ahead with the initial list ordering.